frontend/packages/dev-console/src/components/topology/topology-utils.ts

```diff
@@ -135,7 +134,19 @@ export class TransformTopologyData {
             'metadata.labels',
           ),
           donutStatus: {
-            pods: currentPods,
+            pods: {
+              current: currentPods,
+              previous: previousPods,
+              strategy: deploymentConfig.spec.strategy && deploymentConfig.spec.strategy.type,
+              totalReplicas: deploymentConfig.spec.replicas,
+              currentRCPhase:
+                current &&
+                current.metadata &&
+                current.metadata.annotations['openshift.io/deployment.phase'],
+              currentReplicas: current && current.spec && current.spec.replicas,
+              currentAvailableReplicas:
+                current && current.status && current.status.availableReplicas,
```

Comment:
Why are you not using `get` from `lodash`? By using `get` you wouldn't have to add so many unnecessary checks. For ex - 
```suggestion
                _.get(current, 'status.availableReplicas'),
```